packages/wdio-cucumber-framework/src/index.js

```diff
@@ -0,0 +1,192 @@
+    constructor (cid, config, specs, capabilities, reporter) {
+        this.cwd = process.cwd()
+        this.cid = cid
+        this.specs = specs
+        this.reporter = reporter
+        this.capabilities = capabilities
+        this.config = config
+        this.cucumberOpts = Object.assign(DEFAULT_OPTS, config.cucumberOpts)
+    }
+
+    async run () {
+        let runtimeError
+        let result
+
+        Cucumber.supportCodeLibraryBuilder.reset(this.cwd)
+
+        try {
+            await executeHooksWithArgs(this.config.before, [this.capabilities, this.specs])
+
+            this.registerCompilers()
+            this.loadSpecFiles()
+            this.wrapSteps()
+            Cucumber.setDefaultTimeout(this.cucumberOpts.timeout)
+            const supportCodeLibrary = Cucumber.supportCodeLibraryBuilder.finalize()
+
+            const eventBroadcaster = new EventEmitter()
+            // eslint-disable-next-line no-new
+            new Hookrunner(eventBroadcaster, this.config)
+            const reporterOptions = {
+                capabilities: this.capabilities,
+                ignoreUndefinedDefinitions: Boolean(this.cucumberOpts.ignoreUndefinedDefinitions),
+                failAmbiguousDefinitions: Boolean(this.cucumberOpts.failAmbiguousDefinitions),
+                tagsInTitle: Boolean(this.cucumberOpts.tagsInTitle)
+            }
+
+            new CucumberReporter(eventBroadcaster, reporterOptions, this.cid, this.specs, this.reporter)
+
+            const pickleFilter = new Cucumber.PickleFilter({
+                featurePaths: this.specs,
+                names: this.cucumberOpts.name,
+                tagExpression: this.cucumberOpts.tagExpression
+            })
+            const testCases = await Cucumber.getTestCasesFromFilesystem({
+                cwd: this.cwd,
+                eventBroadcaster,
+                featurePaths: this.specs,
+                order: this.cucumberOpts.order,
+                pickleFilter
+            })
+            const runtime = new Cucumber.Runtime({
+                eventBroadcaster,
+                options: this.cucumberOpts,
+                supportCodeLibrary,
+                testCases
+            })
+
+            await executeHooksWithArgs(this.config.before, [this.capabilities, this.specs])
+            result = await runtime.start() ? 0 : 1
+        } catch (e) {
+            runtimeError = e
+            result = 1
+        }
+
+        await executeHooksWithArgs(this.config.after, [runtimeError || result, this.capabilities, this.specs])
+
+        /**
+         * in case the spec has a runtime error throw after the wdio hook
+         */
+        if (runtimeError) {
+            throw runtimeError
+        }
+        //await reporter.waitUntilSettled()
+
+        return result
+    }
+
+    registerCompilers () {
+        if (!this.cucumberOpts.compiler || this.cucumberOpts.compiler.length === 0) {
+            throw new Error('A compiler must be defined')
```

Comment:
@abjerstedt  are compilers really required? Should it be changed to requireModule instead https://github.com/cucumber/cucumber-js/blob/master/docs/cli.md#transpilation ?

Reply:
Probably not required... no. I originally tried to port the UTs 1:1

Reply:
I used compiler: ['js:fs'] as a workaround for the time being. Looking forward to seeing this merged into the project, so far I've been running tests with no issue on a manually built version of @wdio/sync and the cucumber-framework package